src/parser.rs

```diff
@@ -363,6 +363,36 @@ impl<'a> Parser<'a> {
         Ok(expr)
     }
 
+    pub fn parse_interval_expr(&mut self) -> Result<Expr, ParserError> {
```

Comment:
I wonder if you investigated following a similar pattern to `NOT` which adjusts the precedence based on peeking in the token stream? 

https://github.com/sqlparser-rs/sqlparser-rs/blob/87b4a168cbcd83439c4e8729dd992d150eebd15c/src/parser.rs#L1648-L1660

I couldn't find any example of a similar partten in the codebase

https://github.com/search?q=repo%3Asqlparser-rs%2Fsqlparser-rs%20parse_infix&type=code

What do you think @AugustoFKL ?

Reply:
@alamb I think we may need to refactor the expression parsing... 

With the current structure, we end up with no such good expansion of expressions to parse. And there's still the (big) problem with precedence analysis.

But, for now, and considering the scope of this PR, I think that's OK. Doing it for `NOT` would be better in another PR, right?